chaoskube/chaoskube.go

```diff
@@ -197,7 +200,8 @@ func (c *Chaoskube) DeletePod(victim v1.Pod) error {
 		return nil
 	}
 
-	err := c.Client.CoreV1().Pods(victim.Namespace).Delete(victim.Name, nil)
+	deleteOptions := metav1.DeleteOptions{GracePeriodSeconds: &c.GracePeriod}
```

Comment:
I think there's a difference to specifying a grace period of 30 (the default) to not specifying it at all. It the latter case the `terminationGracePeriodSeconds` field on the pod would be used, hence the pointer value (I'm not sure though).

I would rather do it like that:
* `--grace-period` set to some value bigger than or equal to "0" => value is used
* `--grace-period` not set at all => no value specified in the delete call => Value on Pod is used.

We could use `--grace-period=-1` as the default when the flag is not provided and use `-1` as indicator to leave the `GracePeriodSeconds` in the `DeleteOptions` at `nil` here.

Reply:
I agree. What do you think about:
- `--grace-period` not set => call `Delete(victim.Name, nil)` to use the value on the pod.
- `--grace-period=N` => use this value in `Delete(victim.Name, &deleteOptions)`

This way we avoid using `-1` as a _magic_ value.

Reply:
That would be perfect! I was already thinking about how to code that with `kingpin`. If you can think of a way without using any magic value that would be great.

Reply:
Done. It's not possible to do that with kingpin, so I've implemented your suggestion with the `-1` default value.